lib/libzfs/libzfs_sendrecv.c

```diff
@@ -3122,6 +3122,11 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
 			VERIFY(0 == nvlist_add_uint64(props,
 			    zfs_prop_to_name(ZFS_PROP_CANMOUNT), 0));
 		}
+		if (0 == nvlist_lookup_nvlist(fs, "snapprops",
+			&snapprops_nvlist)) {
```

Comment:
You're right about the non-NULL condition, i just thought that if the first `nvlist_lookup_nvlist` doesn't exit with errors and incorrectly sets `snapprops_nvlist` then we either find `snapname` in the second lookup (overwriting the nvlist with the correct value) or we fail the `VERIFY()`.